### DISCUSSION TITLE
Difference between PushDelegate OnNewToken and  and OnTokenRefreshed

### DISCUSSION THREAD

**Discussion by munkii** [shinyorg/shiny#1535] [Q&A]
What's the difference between PushDelgate.OnNewToken and OnTokenRefreshed. Older versions just had OnTokenRefreshed so i was wondering what changed to prompt the addition of OnNewToken

**Comment by aritchie** [comment]
It was renamed to be more specific that it fires even for new tokens due to a call to RequestAccess.
